tests/checkpointing/test_trainer_checkpoint.py

```diff
@@ -0,0 +1,101 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from argparse import Namespace
+from copy import deepcopy
+import os
+from pathlib import Path
+import pickle
+import platform
+import re
+from unittest import mock
+from unittest.mock import Mock
+
+import cloudpickle
+from omegaconf import Container, OmegaConf
+import pytest
+import torch
+import yaml
+
+import pytorch_lightning as pl
+from pytorch_lightning import seed_everything, Trainer
+from pytorch_lightning.callbacks import Callback, ModelCheckpoint
+from pytorch_lightning.loggers import TensorBoardLogger
+from pytorch_lightning.utilities.cloud_io import load as pl_load
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.base import BoringModel
+import tests.base.develop_utils as tutils
+
+
+def test_finetuning_with_resume_from_checkpoint(tmpdir):
```

Comment:
is this test required anymore? The only change now is that restore is not called while testing, and for that, you have already updated the assertions in the other test.

Reply:
This one is slightly more robust as it makes sure there is a new best_model_path generated. I think we can keep both for now.

Reply:
valid point :+1: